src/main/java/fi/aalto/cs/apluscourses/intellij/utils/Interfaces.java

```diff
@@ -289,4 +296,55 @@ public String getReadNews() {
       return PluginSettings.getInstance().getReadNews();
     }
   }
+
+  public static class SubmissionGroupSelectorImpl implements SubmissionGroupSelector {
+    private static final String JSON_ENTRY_NAME = "lastGroup";
+    private static Cache<String, JSONObject> groupsCache;
+
+    private static void ensureCacheLoaded(@NotNull Project project) {
+      if (groupsCache == null) {
+        Path projectPath = Path.of(Objects.requireNonNull(project.getBasePath()));
+        groupsCache = new JsonFileCache(
+            projectPath.resolve(Path.of(Project.DIRECTORY_STORE_FOLDER, "a-plus-groups.json")));
```

Comment:
It would be nice, I chose the current solution because it was objectively easier to implement. We could go for the way you proposed, but in that case I'd have that as a separate issue and handle both this & duplicate submission checker there